tests/integration/states/test_vmc_networks.py

```diff
@@ -0,0 +1,177 @@
+"""
+    Integration Tests for vmc_networks state module
+"""
+import pytest
+import requests
+from saltext.vmware.utils import vmc_request
+
+
+@pytest.fixture
+def network_id():
+    return "web-tier"
+
+
+@pytest.fixture
+def request_headers(common_data):
+    return vmc_request.get_headers(common_data["refresh_key"], common_data["authorization_host"])
+
+
+@pytest.fixture
+def network_url(common_data, network_id):
+    url = (
+        "https://{hostname}/vmc/reverse-proxy/api/orgs/{org_id}/sddcs/{sddc_id}/"
+        "policy/api/v1/infra/tier-1s/cgw/segments/{network_id}"
+    )
+    api_url = url.format(
+        hostname=common_data["hostname"],
+        org_id=common_data["org_id"],
+        sddc_id=common_data["sddc_id"],
+        network_id=network_id,
+    )
+    return api_url
+
+
+@pytest.fixture
+def network_list_url(common_data):
+    url = (
+        "https://{hostname}/vmc/reverse-proxy/api/orgs/{org_id}/sddcs/{sddc_id}/"
+        "policy/api/v1/infra/tier-1s/cgw/segments"
+    )
+    api_url = url.format(
+        hostname=common_data["hostname"],
+        org_id=common_data["org_id"],
+        sddc_id=common_data["sddc_id"],
+    )
+    return api_url
+
+
+@pytest.fixture
+def common_data(vmc_config):
+    return vmc_config["vmc_nsx_connect"]
+
+
+@pytest.fixture
+def get_networks(common_data, network_list_url, request_headers):
+    session = requests.Session()
+    response = session.get(
+        url=network_list_url,
+        verify=common_data["cert"] if common_data["verify_ssl"] else False,
```

Comment:
Just as a future reference, based on our previous discussion, "common_data" will end out being "opts" - that's very much the pattern that you'll see [in Salt](https://github.com/saltstack/salt/search?q=minion_opts)

Reply:
okay